src/main/java/com/springbootassignment/cruddemo/controller/CourseController.java

```diff
@@ -0,0 +1,65 @@
+package com.springbootassignment.cruddemo.controller;
+
+import com.springbootassignment.cruddemo.entity.Course;
+import com.springbootassignment.cruddemo.entity.Student;
+import com.springbootassignment.cruddemo.service.CourseService;
+import com.springbootassignment.cruddemo.service.StudentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/courses")
+public class CourseController {
+
+    private CourseService courseService;
+
+    @Autowired
+    public CourseController(CourseService theCourseService){
+        courseService = theCourseService;
+    }
+
+    @GetMapping("/list")
+    public String listCourse(Model theModel) {
+        List<Course> theCourses = courseService.findAll();
+        theModel.addAttribute("courses", theCourses);
+        return "list-course";
+    }
+
+
+
+    @GetMapping("/showFormForAdd")
+    public String showFormForAdd(Model model) {
+        Course course = new Course();
+        model.addAttribute("course", course);
+        return "courses/course-form";
+    }
+
+    @PostMapping("/save")
+    public String saveCourse(@ModelAttribute("course") Course course) {
```

Comment:
Don't capture the Entity directly as an Argument instead create DTO for such entities and use Object Mapper to map the dto's to respective entities and vice versa.
You can refer this [link](https://www.baeldung.com/java-dto-pattern#:~:text=When%20to%20Use%20It%3F,roundtrip%20between%20client%20and%20server.)